### PR TITLE
fix: Validate unenrolled TE before importing [DHIS2-17367]

### PR DESCRIPTION
**Pull request by muilpp**

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/enrollment/SecurityOwnershipValidator.java [hunk 1]
```diff
@@ -98,15 +96,43 @@ public void validate(Reporter reporter, TrackerBundle bundle, Enrollment enrollm
...
+  private TrackedEntity getTrackedEntityWhenCreateOrUpdate(
+      TrackerBundle bundle, Enrollment enrollment, Reporter reporter) {
+    TrackedEntity trackedEntity =
+        bundle.getPreheat().getTrackedEntity(enrollment.getTrackedEntity());
+
+    if (trackedEntity == null) {
+      Optional<org.hisp.dhis.tracker.imports.domain.TrackedEntity> te =
+          bundle.findTrackedEntityByUid(enrollment.getTrackedEntity());
+
+      return te.map(
```
Comment (muilpp) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/enrollment/SecurityOwnershipValidator.java:112]: @enricocolasante

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/enrollment/SecurityOwnershipValidator.java [hunk 2]
```diff
@@ -98,15 +96,39 @@ public void validate(Reporter reporter, TrackerBundle bundle, Enrollment enrollm
...
+              entity -> {
+                TrackedEntity newEntity = new TrackedEntity();
+                newEntity.setUid(entity.getUid());
+                newEntity.setOrganisationUnit(
+                    bundle.getPreheat().getOrganisationUnit(entity.getOrgUnit()));
+                return newEntity;
+              })
+          .get();
```
Comment (muilpp) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/enrollment/SecurityOwnershipValidator.java:120]: @enricocolasante is it fine to assume the tracked entity will always be present in the payload when not found in the database?

Comment (enricocolasante) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/enrollment/SecurityOwnershipValidator.java:120]: Yes, we already checked that it exists in [MetaValidator](https://github.com/dhis2/dhis2-core/blob/5da7aa8724d65a3fda83c57afa7bba1962c3d66b/dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/enrollment/MetaValidator.java#L55-L59)

**Comment by muilpp** [comment]
This work will be done in https://github.com/dhis2/dhis2-core/pull/17372

**Pull request closed without merging**